server/lib/gateway/gateway.login.js

```diff
@@ -21,8 +16,6 @@ async function login(email, password) {
     if (this.gladysGatewayClient) {
       this.gladysGatewayClient.disconnect();
     }
-    // create a new instance of the client
-    this.gladysGatewayClient = new this.GladysGatewayClient({ cryptoLib, serverUrl, logger });
```

Comment:
Why removing this from the login function? It's important to create a new client otherwise we risk mixing data from previous login session

I'm not sure I'm super confident on modification on the Gladys Plus code, I can't merge this without super serious tests on my side, and I don't think we want that on this PR 😅

Reply:
> It's important to create a new client otherwise we risk mixing data from previous login session

Wow I didn't imagine that.
I supposed that the client was stateless.

Reply:
The client is entirely stateful as it manages a long-live websocket connections with Gladys plus. 

The disconnect function should remove every state, but just to be safe I prefer to instantiate a new instance 🙂

Reply:
Rollback done